### PR TITLE
Integration tests for OAuth Metadata endpoint

### PR DESCRIPTION
**Pull request by GANGANI**
**Description**
Add Integration tests for the OAuth Metadata endpoint for legacy super tenant mode and legacy tenant mode.

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuthMetadataTestCase.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.wso2.identity.integration.common.utils.ISIntegrationTest;
+import org.wso2.identity.integration.test.utils.DataExtractUtil;
+
+import java.io.IOException;
+
+public class OAuthMetadataTestCase extends ISIntegrationTest {
```
Comment (mefarazath) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuthMetadataTestCase.java:34]: OAuthMetadataTestCase -> OIDCMetadataTest or something similar would give the actual purpose of the TestCase

Comment (GANGANI) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuthMetadataTestCase.java]: Fixed https://github.com/wso2/product-is/pull/8331/commits/b471590aff7777b2d2ec30d21e5d03df071379f1

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OIDCMetadataTest.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        }
+    }
+
+    private HttpResponse sendGetRequest(HttpClient client, String oidcMetadataEndpoint) throws IOException {
+
+        HttpGet getRequest = new HttpGet(oidcMetadataEndpoint);
+        HttpResponse response = client.execute(getRequest);
+        return response;
+    }
+
```
Comment (tharindu-bandara) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OIDCMetadataTest.java:124]: @GANGANI Redundant new line

Comment (GANGANI) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OIDCMetadataTest.java:124]: Fixed in https://github.com/wso2/product-is/pull/9433/commits/d648ac3f71afac10b4880e08d40b4c346058bb4e

**Pull request merged**